internal/cli/branding_texts.go

```diff
@@ -90,164 +117,190 @@ func showBrandingTexts(cli *cli, inputs *brandingTextsInputs) func(cmd *cobra.Co
-		if translations, ok := pv.(map[string]interface{}); ok {
-			for k, v := range translations {
-				if !strings.HasPrefix(k, "error") && !strings.HasPrefix(k, "devKeys") {
-					if _, ok := result[p]; !ok {
-						result[p] = make(map[string]interface{})
-					}
-					if _, ok := result[p][k]; !ok {
-						result[p][k] = make(map[string]interface{})
-					}
-					result[p][k] = v
-				}
+func mergeBrandingTextTranslations(
+	defaultTranslations map[string]interface{},
+	customTranslations map[string]interface{},
+) map[string]map[string]interface{} {
+	mergedTranslations := make(map[string]map[string]interface{})
+
+	for screen, keyTextMap := range defaultTranslations {
+		translations, ok := keyTextMap.(map[string]interface{})
+		if !ok {
+			break
+		}
+
+		for key, text := range translations {
+			if strings.HasPrefix(key, "error") || strings.HasPrefix(key, "devKeys") {
+				continue
+			}
+
+			if _, ok := mergedTranslations[screen]; !ok {
+				mergedTranslations[screen] = make(map[string]interface{})
 			}
+
+			if _, ok := mergedTranslations[screen][key]; !ok {
+				mergedTranslations[screen][key] = make(map[string]interface{})
+			}
+
+			mergedTranslations[screen][key] = text
 		}
 	}
-	for p, pv := range c {
-		if translations, ok := pv.(map[string]interface{}); ok {
-			for k, v := range translations {
-				if _, ok := result[p]; !ok {
-					result[p] = make(map[string]interface{})
-				}
-				if _, ok := result[p][k]; !ok {
-					result[p][k] = make(map[string]interface{})
-				}
-				result[p][k] = v
+
+	for screen, keyTextMap := range customTranslations {
+		translations, ok := keyTextMap.(map[string]interface{})
+		if !ok {
+			break
+		}
+
+		for key, text := range translations {
+			if _, ok := mergedTranslations[screen]; !ok {
+				mergedTranslations[screen] = make(map[string]interface{})
 			}
+
+			if _, ok := mergedTranslations[screen][key]; !ok {
+				mergedTranslations[screen][key] = make(map[string]interface{})
+			}
+
+			mergedTranslations[screen][key] = text
 		}
 	}
-	return result
+
+	return mergedTranslations
+}
+
+func fetchEditedBrandingTextContent(
+	cmd *cobra.Command,
+	cli *cli,
+	inputs *brandingTextsInputs,
+	brandingTextToEdit string,
+) (map[string]interface{}, error) {
+	tempFileName := fmt.Sprintf("%s-prompt-%s.json", inputs.Prompt, inputs.Language)
+
+	err := textBody.OpenEditor(cmd, &inputs.Body, brandingTextToEdit, tempFileName, updateBrandingTextHint(cli))
+	if err != nil {
+		return nil, fmt.Errorf("failed to capture input from the editor: %w", err)
+	}
+
+	var editedBrandingText map[string]interface{}
+	if err := json.Unmarshal([]byte(inputs.Body), &editedBrandingText); err != nil {
+		return nil, err
+	}
+
+	delete(editedBrandingText, textDocsKey)
+
+	return editedBrandingText, nil
+}
+
+func updateBrandingTextHint(cli *cli) func() {
+	return func() {
+		cli.renderer.Infof(
+			"%s once you close the editor, the custom text will be saved. To cancel, CTRL+C.",
+			ansi.Faint("Hint:"),
```

Comment:
Eventually we'll want to abstract hints to avoid copy-pasting `ansi.Faint("Hint:")` everywhere we print a hint. For example, with a `Hint()` renderer method.